### PR TITLE
Ability to clear preferences

### PR DESCRIPTION
**Pull request by nekdenis**
If you do not want keep reference to SharedPreferences this method may be helpful

### CODE REVIEW

File: rx-preferences/src/main/java/com/f2prateek/rx/preferences2/RxSharedPreferences.java [hunk 1]
```diff
@@ -170,4 +170,8 @@ public void onSharedPreferenceChanged(SharedPreferences preferences, String key)
     checkNotNull(defaultValue, "defaultValue == null");
     return new RealPreference<>(preferences, key, defaultValue, StringSetAdapter.INSTANCE, keyChanges);
   }
+  
+  public void clear() {
+      preferences.edit().clear().apply()
```
Comment (NightlyNexus) [inline rx-preferences/src/main/java/com/f2prateek/rx/preferences2/RxSharedPreferences.java]: needs a semicolon. also, two space indent.

**Comment by MyDogTom** [comment]
Looks like more and more people wants be able to delete all data from preferences without keeping reference to `SharedPreferences`. 
Would it be possible to resurrect / accept #38 ? :)

**Comment by f2prateek** [comment]
@MyDogTom #38 was rejected because it instruments it's own custom behaviour which IMO is surprising. Having a passthrough to the standard equivalent SharedPreferences method is fine though.

**Comment by f2prateek** [comment]
Thanks!

**Pull request merged**